### ISSUE TITLE
Install Windows Games

### ISSUE THREAD

**Issue by actraiser30** [Retro-Devils/Mugen-Pi-4#2]
Hello,happy new year,great script

i want to put age of empires 2 to try,how can i do it after your script has the necessary software installed??

Do i put the installed game folder " AOE2 " inside home/pi/retropie/roms/wine/games

I Make a qjoypad layout for the game as described in your info

I Reboot the system 

And the game should be under wine/games/AOE2/ in retropie and i just run it from there??

thanks

**Comment by Retro-Devils** [comment]
Should be that simple .

Your qjoypad layout  has to match exe exactly.

**Comment by actraiser30** [comment]
Thanks
For the mugen type of games now,is the avp mugen supported?

https://alien-vs-predator-mugen.en.uptodown.com/windows

Στις Τρί, 11 Ιαν 2022, 23:29 ο χρήστης Retro-Devils <
***@***.***> έγραψε:

**Comment by Retro-Devils** [comment]
Thier so many mugens no way we can test em all.
They following will cause issues FOR SURE
Winmugens, anything super high resolution, 4v4.

**Comment by Retro-Devils** [comment]
That mugen looks fine

**Comment by symbios24** [comment]
Hi is it possible to create add to your script pre-configured qjoypad layouts for the already tested/working windows games? Like for example age of empires 2?

Thanks

**Comment by Retro-Devils** [comment]
> Hi is it possible to create add to your script pre-configured qjoypad layouts for the already tested/working windows games? Like for example age of empires 2?
> 
> Thanks

I don't understand what you mean . We have just added a host of qjoypad layouts heres .

Issue is where are your games . We store our mugens in .../roms/mugens/games . Wine games in ..../roms/wine/games/.installs

**Comment by symbios24** [comment]
I mean add layouts for specific windows games

**Comment by Retro-Devils** [comment]
> I mean add layouts for specific windows games

Yes those are called qjoypad layouts . We have a bunch upload in this github . Under qjoy layouts

**Comment by symbios24** [comment]
Can you add one for generic strategy games? 
Or just one for age of empires?

**Comment by actraiser30** [comment]
![273577620_483087990047962_5848137629954216361_n](https://user-images.githubusercontent.com/38106933/153709367-75d60d99-df9f-4dec-8597-f0a27829d192.jpg)

i got this error upon compiling in Raspberry Pi 4 1GB

although the compiling/installing still continues is this normal??

**Comment by symbios24** [comment]
Hi, is it possible to add a keyboard layout in qjoypad ?
I don't have a usb joystick or joypad so its not possible to configure the controls to try some windows games
Edit: i managed to find a joypad and configured the keys although there is no save option after i finish i pressed the update and it save them
Upon running the age if empires 2 game i got an error that says "failed to find a layout named Empires2" i press ok and it runs after the game anyway the game has the sound effects but not music is this a bug?

Also i named the Qjoypad layout with the game exe name "Empires2.exe" why it says it cannot find the layout Empires2?

**Comment by Retro-Devils** [comment]
@actraiser30 untested on 1gb pi 4.

@symbios24 it auto looks for a layout with same name as exe . 
What do you mean with game exe. Layouts are located at home/pi/.qjoypad

**Comment by symbios24** [comment]
The game folder is "AgeOfEmpires2" the exe name inside the folder is "Empires2.exe" and the qjoypad layout name is "Empires2.exe" are all this correct?? Or i did something wrong??

**Comment by symbios24** [comment]
Ok i manage to do it,how can i run a specific Window game at system startup? 
And is it possible to delete the desktop after will it affect the game?

**Comment by actraiser30** [comment]
Works perfect in 1Gb PI4

i made a script under will the game work with it?

#!/bin/bash
xset no blank
cd "home/pi/RetroPie/roms/wine/games/aoe2"
qjoypad "empires2" &
wine home/pi/RetroPie/roms/wine/games/aoe2/empires2.exe

thanks

**Comment by Retro-Devils** [comment]
@actraiser30 that script looks fine . Throw in ...../roms/wine/


@symbios24 you want retropie to boot to a wine game? Cause that s a whole different thing lol . The Lyout aoe2.lyt is for aoe2.exe you don't need to add .exe in layout name

**Comment by symbios24** [comment]
Thanks
If i delete the desktop will the game still works after?

**Comment by jajalobrit** [comment]
Hi, I saw that Cuphead can be run but I've tried in several ways and I couldn't.
Trying with a copy of the game files or even installing the game through wine explorer (GOG version) a popup appears when I try to play: "Failed to initialize graphics, make sure you have directx11 installed"
Do I need to do any extra configuration? I already tried to install directx11 and there is always an error in the installation

My attempts made on Raspberry pi 4 with 4GB of ram after installing the script with the option " mugen with all updates" and installing gecko and mono

**Comment by Retro-Devils** [comment]
I would not delete . If you are trying to hide from retropie menu. Rename wine-apps folder .wine-apps the . Will hide from retropie

**Comment by symbios24** [comment]
Im not trying to hide from the menu i just want to save space by deleting the desktop

**Comment by symbios24** [comment]
![IMG_20220217_170838](https://user-images.githubusercontent.com/10866601/154510375-17b2814e-1998-4ea2-aa2d-2cb2921d2aff.jpg)

Im trying to run the game from the command line with the script

#!/bin/bash
xset no blank
cd "/home/pi/RetroPie/roms/wine/games/aoe2"
qjoypad "empires2" &
wine /home/pi/RetroPie/roms/wine/games/aoe2/empires2.exe

But i got this errors in the the photo

I also put the script in the ports and run it within Retropie and got the same error.

Any help?

**Comment by Retro-Devils** [comment]
Did you delete the desktop ? cause that will cause issues.
wheres your aoe2.sh that yo ucan execute from home like that ?

**Comment by actraiser30** [comment]
I did not delete the desktop
The aoe2.sh is in home/pi
What should i do to run it from the script?

Στις Παρ, 18 Φεβ 2022, 16:45 ο χρήστης Retro-Devils <
***@***.***> έγραψε:

**Comment by Retro-Devils** [comment]
If you made this .sh try the following
open terminal 
chmod 755 /home/pi/aoe2.sh
bash /home/pi/aoe2.sh

The chmod 755 makes it executable

**Comment by actraiser30** [comment]
I made the file executable before,but i did not try the bash command

Στις Δευ, 21 Φεβ 2022, 18:39 ο χρήστης Retro-Devils <
***@***.***> έγραψε:

**Comment by Retro-Devils** [comment]
ok

**Comment by symbios24** [comment]
It did not work but i found another way

I tried to run a direct3d game but it failed,is there any special command to run it?

Is there a way to install directx??

 i tried from wine desktop with no luck i got a lot of errors when installing and the installation failed

Is there a way for your script to comes with directx pre-installed or install it from winetricks like mono and gecko?

**Comment by Retro-Devils** [comment]
in theory yes can be preinstalled . But install is easy just try a few versions . 

Directx10 won't work directx9 is highest we've seen work .

**Comment by symbios24** [comment]
I tried a few methods it did not work returned with errors,If you can have direct x9 or direct x7 to be installed with your script from the winetricks section like gecko and mono for example will be great,as many games require it,or if you cant to include a small guide to install it manual
Thanks

**Comment by Retro-Devils** [comment]
So we have just made mono and gecko auto install .

The method is just install Dx9 through desktop we will find the exact one and upload here soon.

**Comment by symbios24** [comment]
Thanks a lot

**Comment by symbios24** [comment]
Question: the game for example age of empires is not always running when i execute the script i have to run it a second time to work and some times it runs with the first time
Is that normal?

**Comment by Retro-Devils** [comment]
Yes this is normal. Takes a few trys unfortunately with some games

**Comment by symbios24** [comment]
Do you know how to run games that require direct3D/Opengl??
I tried with the normal way it did not work got errors that could not find d3d/opengl hardware
Does it require any special switch in the end ??
Thank you

For example wine /game.exe --opengl ??

**Comment by Retro-Devils** [comment]
No special commands id install direct3d but other than that not much I can think of

**Comment by symbios24** [comment]
If you can put in your git page a guide to install directx it will be great

**Comment by Retro-Devils** [comment]
ok i will try and get one together soon . What game are you trying?

**Comment by symbios24** [comment]
Sudden strike, blitzkrieg,gorky17,call of duty 2
And some others

**Comment by Retro-Devils** [comment]
So directx is tall is just the installe.exe  in winetricks they have a few benchmarks . Like 3dmark

**Comment by symbios24** [comment]
Sorry what did you mean,the directx included in wine its just a translation layer i think not the full library.

**Comment by Retro-Devils** [comment]
Ok ok. 
I will try and find a set of links soon and test em . 
They will be here on Github

**Comment by symbios24** [comment]
Thanks a lot
